### PR TITLE
Added subscription into the schema

### PR DESCRIPTION
**Pull request by Ralls0**
# Description

A new feature has been added at graphQL ecosystem. With this change subscriptions are available. Both nodes and pods are supported for subscription this by means of kubectl's watcher.

# How Has This Been Tested?

**To create the namespace for testing**
`kubectl create ns testing`
**To create the pods to watch**
`kubectl apply -f https://k8s.io/examples/controllers/nginx-deployment.yaml -n testing`
***To delete the pods and start from scratch**
`kubectl delete deployment.apps nginx-deployment -n testing`

Into the playgound is possible to perform this graphQL subscription:

```graphql
subscription subscriptionTestA {
   ioK8sApiCoreV1PodUpdate(
    namespace: "testing"
  ) {
    updateType
    payload {
      metadata {
        name
        namespace
      }
      status {
        phase
      }
    } 
  }
}
```

where all pods' changes are shown. If a particular pod want to be subscribed, a `name: <Pod's name>`  argument should be send into the subscription, in this way:

```graphql
subscription subscriptionTestB {
   ioK8sApiCoreV1PodUpdate(
     namespace: "testing",
    name: # Here the Pod's name 
  ) {
    updateType
    payload {
      metadata {
        name
        namespace
      }
      status {
        phase
      }
    } 
  }
}
```

A bug could be seen in the playgound when you test the subscription due to graphql-playground module. This bug is about the scrolling of the responses. At the moment the only way to overcome this issue is to use a browser extension that inject css into the page. A possible extension for fixing that is [Stylus](https://chrome.google.com/webstore/detail/stylus/clngdbkpkpeebahjckkjfobafhncgmne) with the respective code:

```
.graphiql-wrapper > div:first-child > div:first-child > div:nth-child(2) {
    height: 100%;
}
.graphiql-wrapper > div:first-child > div:first-child > div:nth-child(2) > div:nth-child(2) {
    height: 100%;
    overflow-anchor: auto;
}
```

### CODE REVIEW

File: qlkube/src/decorateSubscription.js [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+  const resolvers = {
+    Subscription: {
+      [subscriptionField]: {
+        subscribe: withFilter(
+          () => pubsub.asyncIterator([label]),
+          (payload, variables, info, context) => {
+            return variables.name === undefined
+              ? true
+              : payload.apiObj.metadata.name === variables.name &&
+                  payload.apiObj.metadata.namespace === variables.namespace;
```
Comment (GabriFila) [inline qlkube/src/decorateSubscription.js:50]: The namespaces should always match, this should fix it:
```suggestion
            return payload.apiObj.metadata.namespace === variables.namespace && (variables.name === undefined || payload.apiObj.metadata.name === variables.name);
```

Comment (Ralls0) [reply qlkube/src/decorateSubscription.js:50]: yes, you are right!👍

File: qlkube/src/decorateSubscription.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+  if (!subscriptions) throw 'Parameter subscriptions cannot be empty!';
+  if (!schema) throw 'Parameter schema cannot be empty!';
+
+  let newSchema = schema;
+
+  subscriptions.forEach(e => {
+    newSchema = decorateSubscription(
+      newSchema,
+      e.type,
+      `${e.type}Update`,
```
[the remark this answers is not in the document]
Comment (Ralls0) [reply qlkube/src/decorateSubscription.js:84]: It's ok!

File: qlkube/src/genericFunction.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+function capitalizeType(name) {
```
[the remark this answers is not in the document]
Comment (Ralls0) [reply qlkube/src/genericFunction.js:1]: Perfect

File: qlkube/src/index.js [hunk 4]
```diff
@@ -52,10 +88,26 @@ async function main() {
...
+    console.log(
+      `🚀 Server ready at http://localhost:${PORT}${server.graphqlPath}`
+    );
+    console.log(
+      `🚀 Subscriptions ready at ws://localhost:${PORT}${server.subscriptionsPath}`
+    );
+  });
+
+  try {
+    subscriptions.forEach(e => {
```
[the remark this answers is not in the document]
Comment (Ralls0) [reply qlkube/src/index.js:106]: Yes, it's much more clear!

**Review by GabriFila — APPROVED** [review]

File: qlkube/src/decorateSubscription.js [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+      : 'extend type Subscription';
+
+  // const enumType =
+  //   baseSchema._typeMap.UpdateType === undefined
+  //     ? `enum UpdateType {
+  //   ADDED
+  //   MODIFIED
+  //   DELETED
+  // }`
+  //     : '';
```
Comment (giorio94) [inline qlkube/src/decorateSubscription.js:55]: Commented code should be dropped if not needed

**Review by giorio94 — APPROVED** [review]
/lgtm

Before merging, please squash all commits

**Pull request merged**